README.md

```diff
@@ -621,42 +621,54 @@ object Windows1255StringDeserializer : ResponseDeserializable<String> {
 
 * `basePath` is used to manage common root path. Great usage is for your static API endpoint.
 
-``` Kotlin
+```kotlin
 FuelManager.instance.basePath = "https://httpbin.org"
 ```
 
-``` Kotlin
+```kotlin
 Fuel.get("/get").response { request, response, result ->
     //make request to https://httpbin.org/get because Fuel.{get|post|put|delete} use FuelManager.instance to make HTTP request
 }
 ```
 
-* `baseHeaders` is to manage common HTTP header pairs in format of `Map<String, String>>`.
+* `baseHeaders` is to manage common HTTP header pairs in format of `Map<String, String>`. 
+    - The base headers are only applied if the request does not have those headers set.
 
-``` Kotlin
+```kotlin
 FuelManager.instance.baseHeaders = mapOf("Device" to "Android")
 ```
 
-* `Headers` can be add to a request via the methods `fun header(pairs: Map<String, Any>?): Request`
- or `fun header(vararg pairs: Pair<String, Any>?)`. 
- 
- * The latter method does support multiple values for the same key however as 
- `fun header(pairs: Map<String, Any>?): Request` takes a map this method can not support multiple
-  values for the same key as per the definition of the map.
+* `Headers` can be added to a request via various methods including
+    - `fun header(name: String, value: Any): Request`: `request.header("foo", "a")`
+    - `fun header(pairs: Map<String, Any>): Request`: `request.header(mapOf("foo" to "a"))`
+    - `fun header(vararg pairs: Pair<String, Any>): Request`: `request.header("foo" to "a")`
+    - `operator fun set(header: String, value: Collection<Any>): Request`: `request["foo"] = listOf("a", "b")`
+    - `operator fun set(header: String, value: Any): Request`: `request["foo"] = "a"`
+* By default, all subsequent calls overwrite earlier calls, but you may use the `appendHeader` variant to append values to existing values.
+    - In earlier versions a `mapOf` overwrote, and `varargs pair` did not, but this was confusing. 
 
-``` Kotlin
-Fuel.get("/get").response { request, response, result ->
-    //make request to https://httpbin.org/get with global device header (Device : Android)
-}
+```kotlin
+Fuel.post("/my-post-path")
+    .header("Accept", "text/html, */*; q=0.1")
```

Comment:
Shall we encourage user to use our internal `const val` too? like `CONTENT_TYPE`. if not so we'd better mark them as private?

Reply:
Changed to use our constants!